compiler-rt/lib/rtsan/rtsan_diagnostics.cpp

```diff
@@ -53,35 +47,39 @@ static void PrintStackTrace(uptr pc, uptr bp) {
 }
 
 static void PrintError(const Decorator &decorator,
-                       const DiagnosticsCallerInfo &info) {
-  const char *violation_type = std::visit(
-      Overloaded{
-          [](const InterceptedCallInfo &) { return "unsafe-library-call"; },
-          [](const BlockingCallInfo &) { return "blocking-call"; }},
-      info);
+                       const DiagnosticsInfo &info) {
+  const auto error_type_str = [&info]() -> const char * {
+    switch (info.type) {
+    case DiagnosticsInfoType::InterceptedCall:
+      return "unsafe-library-call";
+    case DiagnosticsInfoType::BlockingCall:
+      return "blocking-call";
+    }
+    return "(unknown error)";
```

Comment:
Do we want to assert or similar? I don't think this should ever happen

Reply:
I agree this shouldn't happen, but I think it's safer not to assert here, just in case a user's runtime is built in debug mode. `clang-tidy` complains about including a `default` case in the switch (because all cases are handled), and I can't think of a better compromise. Happy to take suggestions if you have any ideas!

Reply:
Fine by me :)